libcloud/compute/drivers/outscale.py

```diff
@@ -7948,3 +7949,283 @@ def _to_key_pair(self, key_pair):
+        :rtype: ``bool`` or ``dict``
+        """
+        action = "DeleteMasterKey"
+        data = {"MasterKeyId" : master_key_id, "DaysUntilDeletion" : days_until_deletion, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        if response.status_code == 200:
+            return True
+        return response.json()
+
+    def ex_encrypt_plaintext(
+        self,
+        master_key_id : str,
+        plaintext: str,
+        encryption_context: dict = dict(),
+        dry_run: bool = False,
+    ):
+        """
+        Encrypt a plain key using a CMK.
+
+        :param      master_key_id: This is the master_key_id of the key you are
+        using.
+        :type       master_key_id: ``str``
+
+        :param      plaintext: This is text you are encrypting.
+        :type       plaintext: ``str``
+
+        :param      encryption_context: This is the context you are using.
+        It is described here: https://docs.outscale.com/api#tocsencryptplaintextresponse.
+        :type       encryption_context: ``dict``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: a dict if error else the ciphertext.
+        :rtype: ``dict`` or ``str``
+        """
+        action = "EncryptPlaintext"
+        data = {"MasterKeyId" : master_key_id, "Plaintext" : base64.b64encode(bytes(plaintext, 'utf-8')).decode('ascii'), "EncryptionContext" : encryption_context, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        if response.status_code == 200:
+            return response.json()['Ciphertext']
+        return response.json()
+
+
+    def ex_decrypt_ciphertext(
+        self,
+        ciphertext: str,
+        encryption_context: dict = dict(),
+        dry_run: bool = False,
+    ):
+        """
+        Encrypt a plain key using a CMK.
+
+        :param      ciphertext: This is text you decrypting.
+        :type       ciphertext: ``str``
+
+        :param      encryption_context: This is the context you are using.
+        :type       encryption_context: ``dict``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: a dict if error else the plaintext.
+        :rtype: ``dict`` or ``str``
+        """
+        action = "DecryptCiphertext"
+        data = {"Ciphertext" : ciphertext, "EncryptionContext" : encryption_context, "DryRun": dry_run}
+        response = self._call_api(action, json.dumps(data))
+        if response.status_code == 200:
+            return base64.b64decode(response.json()['Plaintext']).decode('ascii')
+        return response.json()
+
+    def ex_generate_data_key(
+        self,
+        master_key_id : str,
+        size : int,
+        generate_plaintext: bool,
```

Comment:
I think it would be better to have more named parameters as this method need a lot of parameter to be working
I prefere we send back the api error message instead of the library sending an error but can be discussed